core/src/main/scala/kafka/admin/AclCommand.scala

```diff
@@ -62,7 +62,12 @@ object AclCommand {
     var authorizerProperties = Map.empty[String, Any]
     if (opts.options.has(opts.authorizerPropertiesOpt)) {
       val props = opts.options.valuesOf(opts.authorizerPropertiesOpt).asScala.map(_.split("="))
-      props.foreach(pair => authorizerProperties += (pair(0).trim -> pair(1).trim))
+      props.foreach {
+        case Array(key: String, value: String) => authorizerProperties += (key.trim -> value.trim)
+        case array =>
+          println(s"Invalid authorizer property ${array.mkString(",")}")
```

Comment:
We shouldn't let each tool implement its own ad-hoc parsing with its own ad-hoc handling of invalid properties. It is exactly what makes Kafka tools frustrating for users.

If CommanLineUtils.parseKeyValueArgs doesn't have the right error-handling functionality, why not improve it? Sounds like we'll get some improvements to other tools as a result.
(Make sure you find out which tools are affected and document the change)